File: src/main/java/com/libraryapp/controller/BookController.java
package com.libraryapp.controller;

import com.libraryapp.dto.request.BookRequest;
import com.libraryapp.dto.request.BookReturnRequest;
import com.libraryapp.dto.request.LoanBookRequest;
import com.libraryapp.service.BookService;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class BookController {

    private final BookService bookService;

    public BookController(BookService bookService) {
        this.bookService = bookService;
    }

    @PostMapping("/book")
    public Long saveBook(@RequestBody BookRequest request) {
        return bookService.saveBook(request);
    }

    @PostMapping("/book/loan")
    public void loanBook(@RequestBody LoanBookRequest request) {
        bookService.loanBook(request);
    }

    @PutMapping("/book/return")
    public void returnBook(@RequestBody BookReturnRequest request) {
        bookService.returnBook(request);
    }
}

File: src/main/kotlin/com/libraryapp/controller/BookController.kt
package com.libraryapp.controller

import com.libraryapp.domain.user.loanhistory.LoanBookHistoryRepository
import com.libraryapp.domain.user.loanhistory.UserLoanStatus
import com.libraryapp.dto.request.BookRequest
import com.libraryapp.dto.request.BookReturnRequest
import com.libraryapp.dto.request.LoanBookRequest
import com.libraryapp.dto.response.BookStatResponse
import com.libraryapp.service.BookService
import org.springframework.web.bind.annotation.*

@RestController
class BookController(
    private val bookService: BookService,
    private val loanBookHistoryRepository: LoanBookHistoryRepository
) {
    @PostMapping("/book")
    fun saveBook(@RequestBody request: BookRequest): Long {
        return bookService.saveBook(request)
    }

    @GetMapping("/book/loan")
    fun countLanedBook(): Int {
        return bookService.countLoanedBook()
    }

    @PostMapping("/book/loan")
    fun loanBook(@RequestBody request: LoanBookRequest) {
        bookService.loanBook(request)
    }

    @PutMapping("/book/return")
    fun returnBook(@RequestBody request: BookReturnRequest) {
        bookService.returnBook(request)
    }

    @GetMapping("/book/stat")
    fun getBookStatistics(): List<BookStatResponse> {
        return bookService.getBookStatistics()
    }
}
